agent/command/git.go

```diff
@@ -322,31 +307,25 @@ func (c *gitFetchProject) buildSourceCloneCommand(ctx context.Context, comm clie
 		fmt.Sprintf("rm -rf %s", c.Directory),
 	}
 
-	if !opts.fallbackToFullClone {
+	// This sets the branch option for the clone command based on requester only if
+	// this is not a full clone. If this is a full clone, we should clone everything
+	// and fallback.
+	if !opts.fullClone {
 		switch conf.Task.Requester {
+		// TODO: (DEVPROD-6795) Include Git tracking information
+		// inside of the patch document so we can use it here for other requesters.
 		case evergreen.GithubMergeRequester:
-			// If this is a github merge queue task, we can directly clone
-			// the provided branch.
-			if conf.GithubMergeData.HeadBranch != "" {
-				opts.branch = conf.GithubMergeData.HeadBranch
-			} else {
-				// If the head branch is missing, fall back to a full clone.
-				opts.fallbackToFullClone = true
-			}
+			// If this is a GH merge queue task, we can track
+			// the branch from the merge request.
+			opts.branch = conf.GithubMergeData.HeadBranch
 		case evergreen.GithubPRRequester:
-			// If this is a PR task, we can directly clone the provided PR.
-			if conf.GithubPatchData.HeadBranch != "" {
-				opts.branch = conf.GithubPatchData.HeadBranch
-				opts.owner = conf.GithubPatchData.HeadOwner
-				opts.repo = conf.GithubPatchData.HeadRepo
-			} else {
-				// If the head branch is missing, fall back to a full clone.
-				opts.fallbackToFullClone = true
-			}
-		case evergreen.PatchVersionRequester:
-			// TODO: (DEVPROD-6795) Include Git tracking information
-			// inside of the patch document so we can use it here.
-			opts.fallbackToFullClone = true
+			// If this is a PR task, we can track the branch from the PR.
+			opts.branch = conf.GithubPatchData.HeadBranch
+			opts.owner = conf.GithubPatchData.HeadOwner
+			opts.repo = conf.GithubPatchData.HeadRepo
+		case evergreen.RepotrackerVersionRequester:
```

Comment:
i think its helpful to put a comment in this block here so its clear why we're doing this (an empty case statement can look like an accident)